### DISCUSSION TITLE
7.0.13 Release Cycle

### DISCUSSION THREAD

**Discussion by MikeNeilson** [opendcs/opendcs#577] [Announcements]
Please report issues with the release here.

Latest 7.0.13 RC: https://github.com/opendcs/opendcs/releases/tag/7.0.13-RC09

**Comment by MikeNeilson** [comment]
We just corrected an issue with thread handling the would prevent applications from exiting (closing) correctly.

**Reply by MikeNeilson** [reply]
New improvements and some bug fixes.

**Reply by agilmore2** [reply]
One issue I've seen so far:

Sorting of numerical fields is now alphanumeric. After you click on a ID field header to sort it, you can't ever get the original sorting back.

**Reply by MikeNeilson** [reply]
can you name a specific list?
I suspect that might be due to the use of the "SortingTableModel" instead of the JTable DefaultRowSorter. Since we end up actually sorting the table we do in fact loose the original order.

**Reply by agilmore2** [reply]
Every list in compedit for one, and the "hdb" site names are another. Not sure how we handle the site name list, unless there's an option which uses numeric sorts before alpha.

**Reply by MikeNeilson** [reply]
Hmm, and this is new? The only thing Scott and I changed was converting from the table to the model row index so it actually looks up the correct thing.

I may have changed at least one table to use the default sorter, but if you're talking about sorting by a name column, that's always going to be a lexigraphical sorting. 

I did just check a table I was working on and using the default sorting (not the sorting table model) there isn't actually a way to undo sorting, which was... odd to say the least.

I'll have to go look at the code, if it's doing a String::compare type comparison it should behave as you expect as the numbers come before the letters in the ASCII table. So I'm not really sure why it would be doing that.

**Reply by MikeNeilson** [reply]
Oh, I read that wrong, I see it now. Yeah, that's an issue with the TableModel. The tables actually do use Strings for every cell when they shouldn't so now might be the time to actually present the correct classes to the JTable and TableModel.

**Reply by MikeNeilson** [reply]
Yep, doing the exact same thing on the "Select Timeseries" when setting up an input or output for the "Num TS" column.

**Reply by MikeNeilson** [reply]
@hoffmanjs Do you think you could work on this part of the table models? I'm pretty sure it's just making sure getClassAt (I think that's the name) returns an appropriate value. Slight challenge though in that for CWMS, and I may do this for the OpenDCS-Postgres/Oracle database, we may be moving towards UUIDs for the keys instead of numbers. 

Not going to do that tomorrow so don't really need to do anything right now, but there does need to be some consideration between implementations sometimes with those column types.

**Reply by hoffmanjs** [reply]
ok.  I will look into it.

**Comment by ktarbet** [comment]
I installed OpenDCS opendcs-installer-7.0.13-RC08.jar on windows.  When running launcher_start, I get this message below in the console; the app starts ok.   I'm not sure how to respond.

```bat
"The need of this is superseded by using a DCSTOOL_USERDIR for configuration."
"Support for this variable will be moved in a future release. Please update "
"your configuration."
```

**Reply by MikeNeilson** [reply]
Huh, thought I fixed that one. It's related to the CP_SYS_JAR variable, I'm not doing the detection correctly in windows and the message itself is missing some key information. But check the batch file on the latest code. that may have been fixed when that RC.

**Reply by ktarbet** [reply]
Fixed in: https://github.com/opendcs/opendcs/pull/747
